--- bash-5.3.orig/configure	2025-10-30 20:25:54.000000000 +0000
+++ bash-5.3/configure	2025-10-30 20:26:50.000000000 +0000
@@ -23152,8 +23152,7 @@
 		esac ;;
 netbsd*|openbsd*)	LOCAL_CFLAGS="-DDEV_FD_STAT_BROKEN" ;;
 freebsd*|midnightbsd*)	LOCAL_CFLAGS='-DHEREDOC_PIPESIZE=4096' ;;
-*qnx[67]*)	LOCAL_LIBS="-lncurses" ;;
-*qnx*)		LOCAL_CFLAGS="-Dqnx -F -3s" LOCAL_LDFLAGS="-3s" LOCAL_LIBS="-lunix -lncurses" ;;
+*qnx*)          LOCAL_LIBS="-lncurses -lregex" ;;
 powerux*)	LOCAL_LIBS="-lgen" ;;
 cygwin*|msys*)	LOCAL_CFLAGS=-DRECYCLES_PIDS ;;
 opennt*|interix*) LOCAL_CFLAGS="-DNO_MAIN_ENV_ARG -DBROKEN_DIRENT_D_INO -D_POSIX_SOURCE -D_ALL_SOURCE -DRECYCLES_PIDS" ;;
--- bash-5.3.orig/examples/loadables/finfo.c	2025-10-30 20:25:54.000000000 +0000
+++ bash-5.3/examples/loadables/finfo.c	2025-10-30 20:30:47.000000000 +0000
@@ -274,7 +274,7 @@
 	ma = major (st->st_rdev);
 	mi = minor (st->st_rdev);
 #if defined (makedev)
-	d = makedev (ma, mi);
+	d = makedev (0, ma, mi);
 #else
 	d = st->st_rdev & 0xFF;
 #endif
